packages/core/admin/admin/src/pages/SettingsPage/components/Tokens/Table/DeleteButton/tests/DeleteButton.test.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { ThemeProvider, lightTheme } from '@strapi/design-system';
+import { act, fireEvent, render } from '@testing-library/react';
+import { IntlProvider } from 'react-intl';
+import { NotificationsProvider } from '@strapi/helper-plugin';
+
+import DeleteButton from '../index';
+
+jest.mock('@strapi/helper-plugin', () => ({
+  ...jest.requireActual('@strapi/helper-plugin'),
+  useTracking: jest.fn(() => ({ trackUsage: jest.fn() })),
+}));
+
+function getButton(container, name) {
+  return container.querySelector(`button[name="${name}"]`);
+}
+
+function ComponentToTest(props) {
+  return (
+    <IntlProvider locale="en" messages={{}}>
+      <ThemeProvider theme={lightTheme}>
+        <NotificationsProvider toggleNotification={() => {}}>
+          <DeleteButton tokenName="test" onClickDelete={() => {}} {...props} />
+        </NotificationsProvider>
+      </ThemeProvider>
+    </IntlProvider>
+  );
+}
+
+const setup = (props = { onClickDelete: jest.fn() }) => {
+  return render(<ComponentToTest {...props} />);
+};
+
+describe('DeleteButton', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  it('renders and matches the snapshot', () => {
+    const { baseElement } = setup();
+    expect(baseElement).toMatchSnapshot();
+  });
```

Comment:
what benefit does this give us? IMO it would be better to ensure that the labels are "readable" in the document e.g. i should be able to `getByRole('button', { name: deleteButtonName })` which is easier for me, another FE developer to understand whats going wrong if this test breaks when i make changes. Snapshots are so ambiguous and imo not very readable

Reply:
removed the snapshot